controller/src/main/java/io/pravega/controller/task/Stream/StreamMetadataTasks.java

```diff
@@ -371,7 +378,7 @@ private ReaderGroupConfiguration getRGConfigurationFromRecord(final String scope
                      return buildCreateSuccessResponse(scope, rgName);
                  });
          });
-        }, e -> Exceptions.unwrap(e) instanceof RetryableException, READER_GROUP_OPERATION_MAX_RETRIES, executor);
+        }, retryPredicate, READER_GROUP_OPERATION_MAX_RETRIES, executor);
```

Comment:
lets not retry on UNCONDITIONAL predicate. we should fail for exceptions which we know are non retryable.

Reply:
Did not hard code UNCONDITIONAL_PREDICATE here, as at this point I could not see any specific exceptions for which we would not want to retry. When these are found we can change the retryPredicate to include those.
Specific scenarios like - SCOPE_NOT_FOUND are already dealt with earlier in this code.

Reply:
i did not follow your point. by using unconditional predicate here, arent we saying this should be retried on all exceptions?

Reply:
Yes we are. Because at this point we do not know of any exceptions that need to be "non-retryable".
If and when we do, we change the retryPredicate accrodingly for that specific method.

Reply:
but we do know that. for each case we should know which are certainly failing exceptions. that set can be expanded if we realize we havent covered some exception but for most cases we know. 
Things like StoreExceptions.DataNotFound (or DataContainerNotFound) are non retryable for almost all workflows, saving some exceptions.
similarly when we do deletes with conditional flag (delete if empty) then StoreExceptions.DATA_CONTAINS_ELEMENTS is a failing exception. So on and so forth.
based on context we know which are failing cases which should not be retried upon.